.github/actions/buildcocoasdk/action.yml

```diff
@@ -0,0 +1,45 @@
+name: Build Native Dependencies
+runs:
+  using: composite
+
+  steps:
+
+    - name: Get Carthage SHA
+      shell: bash
+      run: echo "CARTHAGE_SHA=$(git rev-parse HEAD)" >> $GITHUB_ENV
+      working-directory: modules/Carthage
+
+    - name: Cache Carthage
+      id: cache-carthage
+      uses: actions/cache@v3
+      with:
+        path: modules/Carthage/.build/debug/carthage
+        key: carthage-${{ env.CARTHAGE_SHA }}
+
+    - name: Build Carthage
+      if: ${{ steps.cache-carthage.outputs.cache-hit != 'true' }}
+      shell: bash
+      run: make all
+      working-directory: modules/Carthage
```

Comment:
Why do we need this? It's available on macOS

Reply:
It's a custom build so we can create xcframeworks for Mac Catalyst properly.  See comments in the `build-sentry-cocoa.sh` script. https://github.com/getsentry/sentry-dotnet/pull/2053/files#diff-6be5d334c8d46a713f0d9a5b40af751ffc613d6e52405e4a0f72fa06c2d91ed6R6-R9

Reply:
(The Carthage submodule is tracking their `xcframework-catalyst` branch.)

Reply:
Happy to kill it if you can find any other solution.  I figure there must be a way to skip Carthage and just use `xcodebuild`, but I lack expertise in this area and couldn't figure it out.

I already looked into using either Swift SPM or CocoaPods, but the matching tooling on the Xamarin side just isn't up to par (docs talk about `sharpie pod`, but that is very broken and MS doesn't seem to be in a hurry to fix it).

However it is done, ultimately we need the build output to look like the following to bind correctly in .NET:

```
.
└── Sentry.xcframework
    ├── Info.plist
    ├── ios-arm64
    │   ├── Sentry.framework
    │   └── dSYMs
    ├── ios-arm64_x86_64-maccatalyst
    │   ├── Sentry.framework
    │   └── dSYMs
    └── ios-arm64_x86_64-simulator
        ├── Sentry.framework
        └── dSYMs
```

We get that by using `carthage build --use-xcframeworks --no-skip-current --platform ios,macCatalyst` but only when using the custom build can we ask for `macCatalyst`.

Only reason this matters is because .NET MAUI uses Mac Catalyst to build macOS apps.